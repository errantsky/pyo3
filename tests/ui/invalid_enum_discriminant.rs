//! As of rust 1.56, enums without #[repr] cannot have discriminant larger than `isize`.
//! The implementation of #[pyclass] depends on this implementation detail.
//! This file tests if this implementation detail is still true.

use pyo3::prelude::*;

#[pyclass]
enum DiscriminantTooLarge{
    Var = 1 << 64,
}

#[pyclass]
enum DiscriminantOverflow{
    Var1 = isize::MAX,
    Overflow,
}

fn main() {}
