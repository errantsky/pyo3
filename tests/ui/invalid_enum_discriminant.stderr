error[E0080]: evaluation of constant value failed
 --> tests/ui/invalid_enum_discriminant.rs:9:11
  |
9 |     Var = 1 << 64,
  |           ^^^^^^^ attempt to shift left by `64_i32`, which would overflow

error[E0370]: enum discriminant overflowed
  --> tests/ui/invalid_enum_discriminant.rs:15:5
   |
15 |     Overflow,
   |     ^^^^^^^^ overflowed on value after 9223372036854775807
   |
   = note: explicitly set `Overflow = -9223372036854775808` if that is desired outcome
